subworkflows/check_and_prepare_input.nf

```diff
@@ -361,17 +360,35 @@ workflow CHECK_AND_PREPARE_INPUT {
                     ch_salmon_index = SALMON_INDEX.out.index
 
                     // Update versions
-                    ch_versions = ch_versions.mix(SALMON_INDEX.out.versions)     
+                    ch_versions = ch_versions.mix(SALMON_INDEX.out.versions)
+                
+                } else {
+                    salmon_index = file(params.salmon_index)
+                    if (salmon_index.exists() && salmon_index.getExtension()=='gz'){
+                        log.info "Compressed salmon index found: decompressing"
+                        // Uncompress
+                        GUNZIP(
+                            salmon_index
+                        )
+                        // Update
+                        salmon_index = GUNZIP.out.decompressed //possible issue here - should this be ch_salmon_index, or do we want it to go through the next steps?
```

Comment:
No, it is correct as it is. We decompress the salmon_index, assign the output to salmon_index and then we check if salmon_index is a directory, and if the directory contains files. If it is so, we assign salmon_index to the ch_salmon_index, otherwise we raise an error.

Reply:
Ah yes i see now!